tests/test_denonavr.py

```diff
@@ -142,74 +146,237 @@ def custom_matcher(self, request: httpx.Request, *args, **kwargs):
+        with asynctest.patch("asyncio.open_connection") as debug_mock:
+            reader = asyncio.StreamReader()
+            debug_mock.return_value = (reader, asyncio.StreamReader())
+            httpx_mock.add_callback(self.custom_matcher)
+
+            self.denon = denonavr.DenonAVR(FAKE_IP)
+            await self.denon.async_setup()
+            await self.denon.async_telnet_connect()
+            mock = Mock()
+            self.future = asyncio.Future()
+            self.denon.register_callback("ALL", mock.method)
+            self.denon.register_callback("ALL", self._callback)
+            reader.feed_data(b"MUON\r")
+            await self.future
+            mock.method.assert_called_once()
+
+    @pytest.mark.asyncio
+    async def test_mute_on(self, httpx_mock: HTTPXMock):
+        """Check that mute on is processed."""
+        with asynctest.patch("asyncio.open_connection") as debug_mock:
+            reader = asyncio.StreamReader()
+            debug_mock.return_value = (reader, asyncio.StreamReader())
+            httpx_mock.add_callback(self.custom_matcher)
+
+            self.denon = denonavr.DenonAVR(FAKE_IP)
+            self.future = asyncio.Future()
+            self.denon.register_callback("ALL", self._callback)
+            await self.denon.async_setup()
+            await self.denon.async_telnet_connect()
+            reader.feed_data(b"MUON\r")
+            await self.future
+            assert self.denon.muted
+
+    @pytest.mark.asyncio
+    async def test_mute_off(self, httpx_mock: HTTPXMock):
+        """Check that mute off is processed."""
+        with asynctest.patch("asyncio.open_connection") as debug_mock:
+            reader = asyncio.StreamReader()
+            debug_mock.return_value = (reader, asyncio.StreamReader())
+            httpx_mock.add_callback(self.custom_matcher)
+
+            self.denon = denonavr.DenonAVR(FAKE_IP)
+            self.future = asyncio.Future()
+            self.denon.register_callback("ALL", self._callback)
+            await self.denon.async_setup()
+            await self.denon.async_telnet_connect()
+            reader.feed_data(b"MUOFF\r")
+            await self.future
+            assert not self.denon.muted
+
+    @pytest.mark.asyncio
+    async def test_power_on(self, httpx_mock: HTTPXMock):
+        """Check that power on is processed."""
+        with asynctest.patch("asyncio.open_connection") as debug_mock:
+            reader = asyncio.StreamReader()
+            debug_mock.return_value = (reader, asyncio.StreamReader())
+            httpx_mock.add_callback(self.custom_matcher)
+
+            self.denon = denonavr.DenonAVR(FAKE_IP)
+            self.future = asyncio.Future()
+            self.denon.register_callback("ALL", self._callback)
+            await self.denon.async_setup()
+            await self.denon.async_telnet_connect()
+            reader.feed_data(b"PWON\r")
+            await asyncio.sleep(0.5)
```

Comment:
This was a mistake. I had done it for all the tests this way before I learned about waiting on the future. I missed rewriting this one. I corrected this test. I'll see what other options there are instead of asynctest but if you have suggestions, unit testing in python is new to me.